src/app/pages/office/cat-b/office.cat-b.page.ts

```diff
@@ -1,58 +1,341 @@
     router: Router,
+    store$: Store<StoreModel>,
+    navController: NavController,
+    toastController: ToastController,
+    modalController: ModalController,
+    outcomeBehaviourProvider: OutcomeBehaviourMapProvider,
+    weatherConditionProvider: WeatherConditionProvider,
+    public questionProvider: QuestionProvider,
+    public alertController: AlertController,
+    private faultCountProvider: FaultCountProvider,
+    private faultSummaryProvider: FaultSummaryProvider,
   ) {
-    super(store$, platform, authenticationProvider, router);
+    super(
+      platform,
+      authenticationProvider,
+      router,
+      store$,
+      navController,
+      toastController,
+      modalController,
+      outcomeBehaviourProvider,
+      weatherConditionProvider,
+    );
+
+    this.showMeQuestions = questionProvider.getShowMeQuestions(TestCategory.B);
+
   }
 
-  ngOnInit() {
+  ngOnInit(): void {
     super.onInitialisation();
+    const currentTest$ = this.store$.pipe(
+      select(getTests),
+      select(getCurrentTest),
+    );
+    const category$ = currentTest$.pipe(
+      select(getTestCategory),
+    );
     this.pageState = {
       ...this.commonPageState,
+      displayShowMeQuestion$: currentTest$.pipe(
+        select(getTestOutcome),
+        withLatestFrom(currentTest$.pipe(
+          select(getTestData),
+          select(getVehicleChecks),
+          select(getShowMeQuestion),
+        )),
+        map(([outcome, question]) =>
+          this.outcomeBehaviourProvider.isVisible(outcome, 'showMeQuestion', question)),
+      ),
+      displayTellMeQuestion$: currentTest$.pipe(
+        select(getTestOutcome),
+        withLatestFrom(currentTest$.pipe(
+          select(getTestData),
+          select(getVehicleChecks),
+          select(getTellMeQuestion),
+        )),
+        map(([outcome, question]) =>
+          this.outcomeBehaviourProvider.isVisible(outcome, 'tellMeQuestion', question)),
+      ),
+      displayDrivingFault$: currentTest$.pipe(
+        select(getTestOutcome),
+        withLatestFrom(currentTest$.pipe(
+          select(getTestData),
+        )),
+        map(([outcome, testData]) =>
+          this.outcomeBehaviourProvider.isVisible(
+            outcome,
+            'faultComment',
+            this.faultSummaryProvider.getDrivingFaultsList(testData, TestCategory.B),
+          )),
+      ),
+      displaySeriousFault$: currentTest$.pipe(
+        select(getTestOutcome),
+        withLatestFrom(currentTest$.pipe(
+          select(getTestData),
+        )),
+        map(([outcome, testData]) =>
+          this.outcomeBehaviourProvider.isVisible(
+            outcome,
+            'faultComment',
+            this.faultSummaryProvider.getSeriousFaultsList(testData, TestCategory.B),
+          )),
+      ),
+      displayDangerousFault$: currentTest$.pipe(
+        select(getTestOutcome),
+        withLatestFrom(currentTest$.pipe(
+          select(getTestData),
+        )),
```

Comment:
if we add the category into the withLatestFrom, this should all be able to live within the base page i'd assume

Reply:
@matthew2564 this covers most all of the 'move to base-page comments' above. I dont want to move the outstanding elements.

Comes from the test-data.cat-b.selector
	displayShowMeQuestion$
	displayTellMeQuestion$
	showMeQuestion$
	tellMeQuestionText$

Needs this.showMeQuestions which is hard keyed to Cat B 
	showMeQuestionOptions$

Uses getTestData which links to CatB reducer
	etaFaults$
	ecoFaults$
	dangerousFaults$
	seriousFaults$
	drivingFaults$
	drivingFaultCount$
	displayDrivingFaultComments$
	displayDrivingFault$
	displaySeriousFault$
	displayDangerousFault$